classes/security/authorization/internal/SubmissionFileStageRequiredPolicy.inc.php

```diff
@@ -1,27 +1,35 @@
 <?php
 /**
- * @file classes/security/authorization/internal/SubmissionFileSubmissionStageRequiredPolicy.inc.php
+ * @file classes/security/authorization/internal/SubmissionFileStageRequiredPolicy.inc.php
  *
  * Copyright (c) 2014-2016 Simon Fraser University Library
  * Copyright (c) 2000-2016 John Willinsky
  * Distributed under the GNU GPL v2. For full terms see the file docs/COPYING.
  *
- * @class SubmissionFileSubmissionStageRequiredPolicy
+ * @class SubmissionFileStageRequiredPolicy
  * @ingroup security_authorization_internal
  *
- * @brief Submission file policy to ensure that we have a file at submission stage.
+ * @brief Submission file policy to ensure that we have a file at a required stage.
  *
  */
 
 import('lib.pkp.classes.security.authorization.internal.SubmissionFileBaseAccessPolicy');
 
-class SubmissionFileSubmissionStageRequiredPolicy extends SubmissionFileBaseAccessPolicy {
+class SubmissionFileStageRequiredPolicy extends SubmissionFileBaseAccessPolicy {
+	/** @var int SUBMISSION_FILE_... */
+	var $_fileStage;
+
+	/** @var boolean Whether the file has to be viewable */
+	var $_viewable;
+
 	/**
 	 * Constructor
 	 * @param $request PKPRequest
 	 */
-	function __construct($request, $fileIdAndRevision = null) {
+	function __construct($request, $fileIdAndRevision = null, $fileStage = null, $viewable = false) {
```

Comment:
Why is `$fileIdAndRevision` optional? OTOH, both that and `$fileStage` should be required parameters.